### PR TITLE
Expose API to fetch cody subscription associated with SAMS identity

### PR DESCRIPTION
**Pull request by thenamankumar**
As part of [#inc-284-plg-users-paying-for-and-being-billed-for-pro-without-being-upgrade](https://sourcegraph.slack.com/archives/C06N8ELPKNH/p1709796118502059) it is noted that users have multiple SAMS accounts associated with their dot-com account which is unexpected.

This PR exposes graphql API to allow us to fetch cody subscription associated with the SAMS external account identity.

Comparing `user.codySubscription` with `user.externalAccounts.nodes.codySubscription` will allow us to debug the mismatch between the SAMS identity using which the user paid for cody pro vs the one they are signed in on dotcom or cody clients.

## Test plan

Run query:
```graphql
{
  user(username: "hereisnaman") {
    codySubscription {
      plan
      status
      currentPeriodStartAt
      currentPeriodEndAt
    }
    externalAccounts {
      nodes {
        serviceID
        serviceType
        accountID
        codySubscription {
          plan
          status
          currentPeriodStartAt
          currentPeriodEndAt
        }
      }
    }
  }
}
```

### CODE REVIEW

File: cmd/frontend/graphqlbackend/external_account.go [hunk 1]
```diff
@@ -51,6 +56,29 @@ func (r *externalAccountResolver) ServiceType() string { return r.account.Servic
...
 func (r *externalAccountResolver) ClientID() string    { return r.account.ClientID }
 func (r *externalAccountResolver) AccountID() string   { return r.account.AccountID }
+
+func (r *externalAccountResolver) CodySubscription(ctx context.Context) (*CodySubscriptionResolver, error) {
+	if !dotcom.SourcegraphDotComMode() {
+		return nil, errors.New("this feature is only available on sourcegraph.com")
+	}
+
+	if r.account.ServiceType == "openidconnect" && r.account.ServiceID == fmt.Sprintf("https://%s", ssc.GetSAMSHostName()) {
+		return nil, nil
```
Comment (burmudar) [inline cmd/frontend/graphqlbackend/external_account.go:66]: Should this no return an error like "ErrNotCodySub" ?

File: cmd/frontend/graphqlbackend/schema.graphql [hunk 2]
```diff
@@ -6844,6 +6844,10 @@ type ExternalAccount implements Node {
     may query this field.
     """
     publicAccountData: PublicExternalAccountData
+    """
+    Cody Subscription associated with the external account.
```
Comment (vdavid) [inline cmd/frontend/graphqlbackend/schema.graphql:6848]: Let's mark this as temporary for debugging an incident.

**Review by vdavid — APPROVED** [review]
Fine after adding the note about the change being temporary.

File: cmd/frontend/graphqlbackend/external_account.go (hunk 1 above)
Comment (thenamankumar) [reply cmd/frontend/graphqlbackend/external_account.go:67]: No, users have non-openidconnect external accounts, for which cody subscription should be nil.

Comment (burmudar) [reply cmd/frontend/graphqlbackend/external_account.go:67]: I think the point I am more conveying is that there should be some value returned. Having both nil seems weird and out of place. So maybe even then `ErrNonOpenIDCodySub`

**Pull request merged**